core/esmf-aspect-model-document-generators/src/main/java/org/eclipse/esmf/aspectmodel/generator/openapi/AspectModelOpenApiGenerator.java

```diff
@@ -697,4 +709,94 @@ private void handleRecursiveSchemas( final ObjectNode aspectSchema, final Object
          }
       }
    }
+
+   static final class ObjectNodeExtension {
+
+      static Function<ObjectNode, ObjectNode> getter( final String propName ) {
+         return node -> (ObjectNode) node.get( propName );
+      }
+
+      static ObjectNode merge( final ObjectNode node, final ObjectNode extension ) {
+         if ( extension == null || extension.isEmpty() ) {
+            return node;
+         }
+         final ObjectNode result = node.deepCopy();
+
+         return mergeObjectNode( result, extension );
+      }
+
+      public static ObjectNode merge( final ObjectNode node, final ObjectNode extension, final String field ) {
+         return ofNullable( extension ).map( getter( field ) ).map( ext -> merge( node.deepCopy(), ext ) ).orElse( node );
+      }
+
+      private static ObjectNode mergeObjectNode( final ObjectNode node, final ObjectNode extension ) {
+         if ( extension == null || extension.isEmpty() ) {
+            return node;
+         }
+         //  cicle for result.fields()
+         extension.fields().forEachRemaining( entry -> {
+            final String key = entry.getKey();
+            final JsonNode value = entry.getValue();
+
+            if ( node.has( key ) ) {
+               final JsonNode resultValue = node.get( key );
+               if ( resultValue.isObject() && value.isObject() ) {
+                  node.set( key, mergeObjectNode( (ObjectNode) resultValue, (ObjectNode) value ) );
+               } else if ( resultValue.isArray() && value.isArray() ) {
+                  node.set( key, mergeArrayNode( (ArrayNode) resultValue, (ArrayNode) value ) );
+               }
+            } else {
+               node.set( key, value );
+            }
+         } );
+         return node;
+      }
+
+      private static ArrayNode mergeArrayNode( final ArrayNode node, final ArrayNode extension ) {
+         ArrayNode result = mergeValueNodeArray( node, extension );
+         result = mergeObjectNodeArray( result, extension );
+         return result;
```

Comment:
Please do not reuse variables if it can be prevented, since it can make following logic difficult. In this case it's easy, but it's not always as straightforward.

```suggestion
         final ArrayNode arrayNode = mergeValueNodeArray( node, extension );
         return mergeObjectNodeArray( arrayNode, extension );
```

Reply:
Here, I intentionally used a sequence of calls to increase readability, but it can be improved